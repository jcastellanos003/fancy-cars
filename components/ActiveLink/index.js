import { useRouter } from 'next/router'
import Link from 'next/link';

const ActiveLink = ({ children, ...props }) => {
  const router = useRouter();
  const child = React.Children.only(children);
  
  return (
    <Link {...props}>
      {React.cloneElement(child, { active: router.pathname === props.href, theme: props.theme })}
    </Link>
  )
};

export default ActiveLink;
