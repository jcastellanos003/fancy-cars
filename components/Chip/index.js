import PropTypes from 'prop-types';

import * as components from './styles';

export const Chip = ({ name, onClose }) => (
  <components.StyledChip>
    <span>{name}</span> <components.CloseIcon onClick={onClose} />
  </components.StyledChip>
);

Chip.propTypes = {
  name: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired
};
