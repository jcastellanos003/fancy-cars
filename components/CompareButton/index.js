import * as components from './styles';

export const CompareButton = () => (
  <components.StyledCompareButton>
    <components.CompareButtonIcon />
  </components.StyledCompareButton>
);
