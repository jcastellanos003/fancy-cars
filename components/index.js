export * from './ActiveLink';
export * from './Nav';
export * from './Meta';
export * from './SearchBox';
export * from './Card';
export * from './VehicleCard';
export * from './BackButton';
export * from './PageHeader';
export * from './Chip';
export * from './CompareButton';
export * from './PageTitle';
export * from './PriceLabel';
