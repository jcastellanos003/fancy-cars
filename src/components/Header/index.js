import { Nav } from '../Nav';
import * as styled from './styles';

export const Header = () => (
  <section>

    <styled.CustomHeader>
      <styled.Logo src="/logo.png"/>
    </styled.CustomHeader>

    <Nav />

  </section>
);
