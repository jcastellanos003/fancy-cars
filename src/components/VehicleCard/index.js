import PropTypes from 'prop-types';

import { Card, CardHeader, CardContent  } from '../Card';
import { VehicleImage } from './styles';

export const VehicleCard = ({
    image,
    size,
    children
  }) => (
    <section>
      <Card size={size}>

        <CardHeader>
          <VehicleImage src={image} />
        </CardHeader>

        <CardContent>
          {children}
        </CardContent>

      </Card>
    </section>
);

VehicleCard.propTypes = {
  size: PropTypes.oneOf(['medium', 'large']).isRequired,
  image: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
