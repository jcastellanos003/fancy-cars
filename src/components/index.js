export * from './ActiveLink/ActiveLink';
export * from './BackButton/BackButton';
export * from './Card';
export * from './CompareButton/CompareButton';
export * from './CompareLink/CompareLink';
export * from './Chip/Chip';
export * from './Header/Header';
export * from './HorizontalPane/HorizontalPane';
export * from './InnerPage/InnerPage';
export * from './Master/Master';
export * from './Meta/Meta';
export * from './Nav/Nav';
export * from './PageHeader/PageHeader';
export * from './PageTitle/PageTitle';
export * from './PriceLabel/PriceLabel';
export * from './SearchBox/SearchBox';
export * from './SelectedCard/SelectedCard';
export * from './SelectedButton/SelectedButton';
export * from './VehicleCard/VehicleCard';
