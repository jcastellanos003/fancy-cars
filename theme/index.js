export * from './DefaultTheme';
export * from './GlobalStyles';
