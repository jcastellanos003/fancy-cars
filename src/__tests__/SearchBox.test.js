import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { SearchBox } from '../components';
import { theme } from '../theme';

describe('SearchBox Component', () => {
  test('Matches the snapshot', () => {
    const handleSubmit = event => event.preventDefault();
    const handleChange = event => event.preventDefault();
    const searchBox = create(
      <ThemeProvider theme={theme}>
        <SearchBox onSubmit={handleSubmit} onChange={handleChange} />
      </ThemeProvider>
      );

    expect(searchBox.toJSON()).toMatchSnapshot();
  })
})
