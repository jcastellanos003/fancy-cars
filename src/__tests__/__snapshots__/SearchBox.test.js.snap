// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component Matches the snapshot 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  width: 1.3rem;
}

.c2 {
  color: #FAFAFA;
  background: #2A293B;
  border: 1px solid #2A293B;
  border: none;
  border-top-right-radius: 3px;
  border-bottom-right-radius: 3px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 85%;
  margin: 0 auto;
}

.c1 {
  background: #FAFAFA;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  border: 1px solid #2A293B;
  padding: 10px;
  font-size: 0.8rem;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  outline: none;
}

<form
  autoComplete="off"
  className="c0"
  onSubmit={[Function]}
>
  <input
    className="c1"
    name="search"
    onChange={[Function]}
    placeholder="Search your car"
    required={true}
    type="text"
  />
  <button
    className="c2"
    type="submit"
  >
    <svg
      aria-hidden="true"
      className="c3 c4"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
      <path
        d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 00-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
      />
    </svg>
  </button>
</form>
`;
